src/local_scheduler/local_scheduler_algorithm.cc

```diff
@@ -575,10 +582,10 @@ int fetch_object_timeout_handler(event_loop *loop, timer_id id, void *context) {
  */
 void dispatch_tasks(LocalSchedulerState *state,
                     SchedulingAlgorithmState *algorithm_state) {
-  task_queue_entry *elt, *tmp;
-
   /* Assign as many tasks as we can, while there are workers available. */
-  DL_FOREACH_SAFE(algorithm_state->dispatch_task_queue, elt, tmp) {
+  auto it = algorithm_state->dispatch_task_queue->begin();
+  while (it != algorithm_state->dispatch_task_queue->end()) {
```

Comment:
this could be written as a for loop:
`for (auto it = algorithm_state->dispatch_task_queue->begin(); 
         it != algorithm_state->dispatch_task_queue->end();) {
`